pkg/handler/processor/guesser/guesser_test.go

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guesser
+
+import (
+	"context"
+	"testing"
+
+	testdata "github.com/guacsec/guac/internal/testing/processor"
+	"github.com/guacsec/guac/pkg/handler/processor"
+)
+
+func Test_GuessDocument(t *testing.T) {
+	testCases := []struct {
+		name     string
+		document *processor.Document
+		expected processor.DocumentType
+	}{{
+		name: "DocumentUnknown",
+		document: &processor.Document{
+			Blob:              []byte(`{ "abc": "def"}`),
+			Type:              processor.DocumentUnknown,
+			Format:            processor.FormatUnknown,
+			SourceInformation: processor.SourceInformation{},
+		},
+		expected: processor.DocumentUnknown,
```

Comment:
Would be great if we can add the check for the format as well! Maybe `expectedType` and `expectedFormat`!